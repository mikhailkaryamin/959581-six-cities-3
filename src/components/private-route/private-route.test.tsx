import * as React from 'react';
import * as renderer from 'react-test-renderer';
import configureStore from "redux-mock-store";
import {
  Provider
} from "react-redux";
import {
  BrowserRouter
} from 'react-router-dom';
import {
  Store
} from '../../mocks/initialStateTest';
import PrivateRoute from './private-route';

const mockStore = configureStore([]);
const MockComponent = <div />;
const render = () => MockComponent;

test(`Private route render correctly`, () => {
  const store = mockStore(Store.WITH_AUTH);
  const tree = renderer
      .create(
          <Provider
            store={store}
          >
            <BrowserRouter>
              <PrivateRoute
                exact={true}
                path={``}
                render={render}
              />
            </BrowserRouter>
          </Provider>
      ).toJSON();
  expect(tree).toMatchSnapshot();
});
