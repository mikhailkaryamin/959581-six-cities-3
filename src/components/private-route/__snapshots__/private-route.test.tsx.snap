// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Private route render correctly 1`] = `<div />`;
